### PR TITLE
INSP: Don't register problems with `INFORMATION` level in batch mode

### LINKED ISSUE #9498: Unused Import Inspection has wrong level type
## Environment

* **IntelliJ Rust plugin version:** 0.4.180.4931-222-beta
* **Rust toolchain version:** 1.64.0 (a55dd71d5 2022-09-19) x86_64-apple-darwin
* **IDE name and version:** CLion 2022.2.3 (CL-222.4167.35)
* **Operating system:** macOS 12.5.1
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.cargo.features.settings.gutter, org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description
Exception: 

<details>

```
com.intellij.diagnostic.PluginException: Tool #RsUnusedImport (class org.rust.ide.inspections.lints.RsUnusedImportInspection) registers 'INFORMATION'-level problem in batch mode on FILE. Warnings of the 'INFORMATION' level are invisible in the editor and should not become visible in batch mode. Moreover, since 'INFORMATION'-level fixes act more like intention actions, they could e.g. change semantics and thus should not be suggested for batch transformations [Plugin: org.rust.lang]
	at com.intellij.codeInspection.InspectionEngine$2.registerProblem(InspectionEngine.java:261)
	at com.intellij.codeInspection.ProblemsHolder.registerProblem(ProblemsHolder.java:52)
	at org.rust.ide.inspections.RsProblemsHolder.registerProblem(RsLocalInspectionTool.kt:86)
	at org.rust.ide.inspections.lints.RsLintInspection.registerLintProblem(RsLintInspection.kt:34)
	at org.rust.ide.inspections.lints.RsUnusedImportInspection.markAsUnused(RsUnusedImportInspection.kt:100)
	at org.rust.ide.inspections.lints.RsUnusedImportInspection.markUnusedSpecks(RsUnusedImportInspection.kt:88)
	at org.rust.ide.inspections.lints.RsUnusedImportInspection.access$markUnusedSpecks(RsUnusedImportInspection.kt:30)
	at org.rust.ide.inspections.lints.RsUnusedImportInspection$buildVisitor$1.visitUseItem(RsUnusedImportInspection.kt:54)
	at org.rust.lang.core.psi.impl.RsUseItemImpl.accept(RsUseItemImpl.java:27)
	at org.rust.lang.core.psi.impl.RsUseItemImpl.accept(RsUseItemImpl.java:32)
	at com.intellij.codeInspection.InspectionEngine.acceptElements(InspectionEngine.java:81)
	at com.intellij.codeInspection.InspectionEngine.createVisitorAndAcceptElements(InspectionEngine.java:56)
	at com.intellij.codeInspection.InspectionEngine.lambda$inspectElements$9(InspectionEngine.java:282)
	at com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:136)
	at com.intellij.concurrency.ApplierCompleter.execAndForkSubTasks(ApplierCompleter.java:149)
	at com.intellij.openapi.application.impl.ApplicationImpl.tryRunReadAction(ApplicationImpl.java:1154)
	at com.intellij.concurrency.ApplierCompleter.lambda$wrapInReadActionAndIndicator$1(ApplierCompleter.java:92)
	at com.intellij.openapi.progress.impl.CoreProgressManager.lambda$executeProcessUnderProgress$12(CoreProgressManager.java:608)
	at com.intellij.openapi.progress.impl.CoreProgressManager.registerIndicatorAndRun(CoreProgressManager.java:683)
	at com.intellij.openapi.progress.impl.CoreProgressManager.computeUnderProgress(CoreProgressManager.java:639)
	at com.intellij.openapi.progress.impl.CoreProgressManager.executeProcessUnderProgress(CoreProgressManager.java:607)
	at com.intellij.openapi.progress.impl.ProgressManagerImpl.executeProcessUnderProgress(ProgressManagerImpl.java:60)
	at com.intellij.concurrency.ApplierCompleter.wrapInReadActionAndIndicator(ApplierCompleter.java:104)
	at com.intellij.concurrency.ApplierCompleter.compute(ApplierCompleter.java:86)
	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:754)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)

```

</details>

## Steps to reproduce

 Open some project
Run inspections on the whole project: `Inspect code` 
image

### PR DESCRIPTION
**Pull request by dima74**
Fixes #9498

This should not affect users (problems with `INFORMATION` level are filtered out in batch mode anyway)

### CODE REVIEW

File: src/main/kotlin/org/rust/ide/inspections/RsLocalInspectionTool.kt [hunk 1]
```diff
@@ -106,10 +106,13 @@ class RsProblemsHolder(private val holder: ProblemsHolder) {
         rangeInElement: TextRange,
         vararg fixes: LocalQuickFix?
     ) {
-        if (element.existsAfterExpansion) {
+        if (element.existsAfterExpansion && checkHighlightType(highlightType)) {
             holder.registerProblem(element, message, highlightType, rangeInElement, *fixes)
         }
     }
+
+    private fun checkHighlightType(highlightType: ProblemHighlightType): Boolean =
```
Comment (Undin) [inline src/main/kotlin/org/rust/ide/inspections/RsLocalInspectionTool.kt:114]: Maybe come up with a better name? right now it's not clear what it does

Comment (dima74) [reply src/main/kotlin/org/rust/ide/inspections/RsLocalInspectionTool.kt]: Replaced with `showProblemWithType`

Comment (Undin) [reply src/main/kotlin/org/rust/ide/inspections/RsLocalInspectionTool.kt]: Now, the name tells me that method should show some problem but actually, it doesn't show anything

I suggest something like `isAnnotationWithTypeAllowed`

**Review by Undin — COMMENTED** [review]
Did you intentionally skip some `registerProblem` overloads?

**Review by Undin — APPROVED** [review]
bors r+

**Pull request merged**